src/react/databrowser/objects/details/ObjectRestorationButtonAndModal.tsx

```diff
@@ -0,0 +1,185 @@
+          </tr>
+        </thead>
+        <tbody>
+          <tr style={{ borderBottom: `1px solid ${theme.brand.border}` }}>
+            <td align="left">{objectMetadata.objectKey}</td>
+            <td align="left">
+              {formatShortDate(new Date(objectMetadata.lastModified))}
+            </td>
+            <td align="left">
+              <PrettyBytes bytes={objectMetadata.contentLength} decimals={2} />
+            </td>
+            <td align="left">{objectMetadata.storageClass || 'default'}</td>
+          </tr>
+        </tbody>
+      </table>
+    );
+  };
+  const RestorationModalContent = () => {
+    return (
+      <div style={{ maxWidth: '45rem' }}>
+        {restoreError && (
+          <Banner
+            variant="warning"
+            title="Restore Object Failed"
+            icon={<Icon name="Exclamation-circle" color="statusWarning"></Icon>}
+          >
+            {restoreError}
+          </Banner>
+        )}
+        <Stack direction="vertical" gap="f32">
+          <Text>
+            To restore an object, you need to launch a Restoration request, then
+            wait for the object to become available.
+          </Text>
+          <ObjectTable />
+          <Stack>
+            Duration of Restoration
+            <Input
+              id="restorationDuration"
+              name="restorationDuration"
+              type="number"
+              value={restorationDays}
+              onChange={(e: ChangeEvent<HTMLInputElement>) =>
+                setRestorationDays(parseInt(e.target.value))
+              }
+              size="1/3"
+              min={1}
+            ></Input>{' '}
+            day{restorationDays >= 2 ? '(s)' : ''}
+          </Stack>
+          <Text>
+            A temporary copy of your object will be available during a limited
+            number of days. After this period, the restored copy will be
+            automatically deleted.
+          </Text>
+        </Stack>
+      </div>
+    );
+  };
+
+  const restore = useMutation(
+    async () => {
+      await zenkoClient.restoreObject(
+        bucketName,
+        objectMetadata.objectKey,
+        restorationDays,
+        objectMetadata.versionId,
+      );
+    },
+    {
+      onSuccess: () => {
+        handleClose();
+      },
+      onError: (err: AWSError) => {
+        setRestoreError(err.message);
+      },
+    },
+  );
+
+  const isObjectRestoredOrOnGoing =
+    objectMetadata.restore?.ongoingRequest ||
+    !!objectMetadata.restore?.expiryDate;
+
+  return (
+    <>
+      <Button
+        id="restore-button"
+        type="button"
+        variant="outline"
+        disabled={isObjectRestoredOrOnGoing || restore.isLoading}
+        label="Restore"
+        onClick={() => setIsModalOpen(true)}
+        tooltip={
+          isObjectRestoredOrOnGoing
+            ? {
+                overlay:
+                  'The object is already restored or the restoration is ongoing, hence disable the button.',
```

Comment:
Nit: rewording proposal:
`The button is disabled because the object is already restored or the restoration is ongoing`